### PR TITLE
Fix logical replication initialization when subscriber side is a multi-node cluster.

### PR DESCRIPTION
**Pull request by seut**
- Fix logical replication initialization, must only run on the master node.
- Renew retention lease while polling remote shard changes 
- Remove differentiation of publisher and subscriber shardId internally 

See commits for details.

### CODE REVIEW

File: server/src/main/java/io/crate/replication/logical/ShardReplicationService.java [hunk 1]
```diff
@@ -68,27 +78,74 @@ public void close() throws IOException {
         }
     }
 
+    void getRemoteClusterClient(Index index, Consumer<Client> onSuccess, Consumer<Exception> onFailure) {
+        var subscriptionName = subscriptionName(index.getUUID());
```
Comment (mkleen) [inline server/src/main/java/io/crate/replication/logical/ShardReplicationService.java:86]: Minor: I find `subscriptionName` in this context confusing. We have subscriptions which can have a name such as `sub1` but this here is the name of the remote cluster, right ?

Comment (seut) [reply server/src/main/java/io/crate/replication/logical/ShardReplicationService.java:86]: > but this here is the name of the remote cluster, right ?

No it's not the name of the remote cluster, we register connections based on the subscription name while e.g. OpenSearch register connections based on the remote cluster name.

File: server/src/main/java/io/crate/replication/logical/ShardReplicationChangesTracker.java [hunk 2]
```diff
@@ -277,11 +281,31 @@ private void updateBatchFetched(boolean success,
             }
         }
 
-        // schedule next poll
-        cancellable = threadPool.schedule(
-            this::requestBatchToFetch,
-            replicationSettings.pollDelay(),
-            ThreadPool.Names.LOGICAL_REPLICATION
+        // Renew retention lease with global checkpoint so that any shard that picks up shard replication task
```
Comment (mkleen) [inline server/src/main/java/io/crate/replication/logical/ShardReplicationChangesTracker.java:284]: Shards also have local checkpoints, which describe the lag the shard. I guess global checkpoint is safe to use, but could that be less efficient (e.g. data processed twice) ?

**Review by mkleen — APPROVED** [review]
👍 Great work. Only 2 minor points.

File: server/src/main/java/io/crate/replication/logical/ShardReplicationService.java (hunk 1 above)
Comment (mkleen) [reply server/src/main/java/io/crate/replication/logical/ShardReplicationService.java:86]: 👍

**Comment by mkleen** [comment]
https://jenkins.crate.io/job/CrateDB/job/crate_on_pr/15938/testReport/junit/io.crate.integrationtests/LogicalReplicationITest/test_subscribing_to_publication_while_table_exists_raises_error/

File: server/src/main/java/io/crate/replication/logical/ShardReplicationChangesTracker.java (hunk 2 above)
Comment (mkleen) [reply server/src/main/java/io/crate/replication/logical/ShardReplicationChangesTracker.java:284]: See https://github.com/opendistro-for-elasticsearch/cross-cluster-replication/blob/eebceb244d7456b2c675b8e6765f3cbccd316a9b/src/main/kotlin/com/amazon/elasticsearch/replication/task/shard/ShardReplicationTask.kt#L116

Comment (seut) [reply server/src/main/java/io/crate/replication/logical/ShardReplicationChangesTracker.java:284]: Using the local checkpoint isn't safe to use here as it does not reflect the current "safe" state of all shards on the publishing cluster. Consider a failure on the primary shard with a local checkpoint 10, while the replica's local checkpoint and global checkpoint is still on 9. If the the local checkpoint 10 would be used and already replicated to any subscriber, but on the publisher side the repopulated (global checkpoint) is still on 9, an invariant state would have been replicated.

Opendistro's CCR seems to be outdated and afaik is superseded by OpenSearch's CCR which also (correctly) do not use an unsafe local checkpoint: https://github.com/opensearch-project/cross-cluster-replication/blob/main/src/main/kotlin/org/opensearch/replication/task/shard/ShardReplicationTask.kt#L204

Comment (mkleen) [reply server/src/main/java/io/crate/replication/logical/ShardReplicationChangesTracker.java:284]: 👍

Comment (mkleen) [reply server/src/main/java/io/crate/replication/logical/ShardReplicationChangesTracker.java:284]: This makes totally sense.

**Pull request merged**